style-guide.rst

```diff
@@ -0,0 +1,628 @@
+
+.. _title-verb-forms:
+
+Verb forms
+-----------
+
+If a document or section describes a procedure that someone might do, use a gerund. Do not use the "How to..." construction.
+
+.. rubric:: Wrong
+
+.. code-block:: rst
+
+  How to install ODK Collect
+  --------------------------
+    
+.. rubric:: Right
+
+.. code-block:: rst
+
+  Installing ODK Collect
+  ----------------------
+    
+If section title is a directive to do something (for example, as a step in a procedure), use an imperative. 
+
+.. code-block:: rst
+
+  Installing ODK Aggregate
+  ------------------------
+  
+  Download ODK Aggregate
+  ~~~~~~~~~~~~~~~~~~~~~~
+
+  Section content here.
+
+  
+.. _section-label-style-guide:  
+  
+Section labels
+~~~~~~~~~~~~~~~~
+
+Section titles should almost always be preceded by labels.
+
+The only exception is very short subsections that repeat --- like the **Right** and **Wrong** titles in this document or the **XLSForm Rows** and **XForm XML** sections in the :doc:`form-widgets` document.
+
+In these cases, you may want to use the :rst:dir:`rubric` directive.
+    
+.. _other-title-considerations:
+      
+Other titling considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Do not put step numbers in section titles.
+- Readers skim. Section titles should be clear and provide information.
+
+  
+.. _writing-about-code:
+  
+Writing code and writing about code
+======================================
+
+ODK Documentation includes code samples in a number of languages. Make sure to follow generally accepted coding style for each language. 
+
+.. _indenting:
+
+Indenting
+------------
+
+In code samples:
+
+- Use spaces, not tabs.
+- Two spaces for logical indents in most languages.
+
+  - Python samples must use `four spaces per indent level <https://www.python.org/dev/peps/pep-0008/#indentation>`_.
+  
+- Strive for clarity. Sometimes nonstandard indentation, especially when combined with non-syntactic line breaks, makes things easier to read.
+
+  - Make sure that line breaks and indentation stay within the valid syntax of the language.
+
+Using two spaces keeps code sample lines shorter, which makes them easier to view.
+
+
+.. rubric:: Example of indenting for clarity
+
+.. code-block:: HTTP
+
+  HTTP/1.0 401 Unauthorized
+  Server: HTTPd/0.9
+  Date: Sun, 10 Apr 2005 20:26:47 GMT
+  WWW-Authenticate: Digest realm="testrealm@host.com",
+			   qop="auth,auth-int",
+			   nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093",
+			   opaque="5ccc069c403ebaf9f0171e9517f40e41"
+  Content-Type: text/html
+  Content-Length: 311
+
+.. _meaningful-names:
+
+Meaningful names
+-----------------
+
+When writing sample code, avoid meaningless names.
+
+.. rubric:: Wrong
+
+.. code-block:: python
+
+  def myFunction(foo):
+
+    for bar in foo:
+       bar[foo] = foo[spam] + spam[foo]
+
+    return foobar
+
+.. _xml-html-style-guide:
+
+XML and HTML
```

Comment:
We don't really use HTML in ODK, so I don't think the HTML section is relevant. XML, is fine.

Reply:
ok

Reply:
well...

There's almost nothing here that is HTML specific (except the one short section on null tags), so I don't see the harm in leaving it in. Also, it might come up in docs about docs templating.